### ISSUE TITLE
Error Code 32

### ISSUE THREAD

**Issue by DeepS1X** [MyTwitter/MyTwitter#16]
PS G:\New Folder\MD> Send-Tweet -Message "Hello, world" -verbose
VERBOSE: Generated Oauth none string 'NjM2Mjc2NDc0OTUwNTAwMTI5MTIzNDUg'
VERBOSE: Generated epoch time '17269.06:31:35.0532169'
VERBOSE: Generated Oauth timestamp '1492065095'
VERBOSE: Base signature generated 'POST&https%3A%2F%2Fapi.twitter.com%2F1.1%2Fstatuses%2Fupdate.json&oauth_consumer_key%3Di3Bv4pGlxukSuLeUOlMJxCjBp%26oauth_nonce%3DNjM2Mjc2NDc0OTUwNTAwMTI5MTIzNDUg%26oauth_signature_method%3DHMAC-SHA1%26oauth_timestamp%3D1
492065095%26oauth_token%3D851875058301767680-5a4UwftJvQfJw3POT8tgNueKePLXPPh%26oauth_version%3D1.0%26status%3DHello%2C%20world'
VERBOSE: Using signature '5AflZfplaU9a1moGd1KTlqAodIU='
VERBOSE: Using authorization string 'OAuth oauth_consumer_key="i3Bv4pGlxukSuLeUOlMJxCjBp", oauth_nonce="NjM2Mjc2NDc0OTUwNTAwMTI5MTIzNDUg", oauth_signature="5AflZfplaU9a1moGd1KTlqAodIU%3D", oauth_signature_method="HMAC-SHA1", oauth_timestamp="1492065095", 
oauth_token="851875058301767680-5a4UwftJvQfJw3POT8tgNueKePLXPPh", oauth_version="1.0"'
VERBOSE: Using POST body 'status=Hello, world'
VERBOSE: POST https://api.twitter.com/1.1/statuses/update.json with -1-byte payload
Invoke-RestMethod : {"errors":[{"code":32,"message":"Could not authenticate you."}]}
At line:187 char:3
+         Invoke-RestMethod -URI $HttpEndPoint -Method Post -Body $Body ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand

So, I can get Get-TweetTimeline to work fine, but Twitter absolutely refuses to allow me to use Send-Tweet or DM.

Any ideas on what the problem is?

**Comment by adbertram** [comment]
Did you make sure you created your Twitter app exactly like what's described in this blog post? http://www.adamtheautomator.com/twitter-powershell/

**Comment by DeepS1X** [comment]
I did. However, I found a different framework which allows me to call any APIs that works just fine now.

**Comment by DeepS1X** [comment]
So I wouldn't worry too much about this if no one else is reporting the issue.

**Comment by dorsy99** [comment]
Hey, I'm getting this as well, I think it *might* have to do with commas in your tweet. I read somewhere there were issues with !'s as well.

**Comment by DeepS1X** [comment]
Yeah, I could see that being an issue if it weren't URL encoded.
But knowing REST APIs, their error codes are usually spot on. If I had to
guess anything, it's that something is probably screwing with the
authentication procedure.

**Comment by dorsy99** [comment]
@DeepS1X Yeah, but it works perfectly posting without a comma, and as soon as I add one I get that error. I've tried url-encoding my string before sending it and still no dice.

**Comment by DeepS1X** [comment]
I think I've figured it out. The comma is the delimiter between the content
and authentication sections of the HTTP request. If you URL encoded the
command, "Hello, World", it should work. Otherwise, it thinks "World" is
the authentication field and disregards the actual authentication field.

That's my hypothesis.

**Comment by dorsy99** [comment]
Ok, I've just done some testing and adding the extra characters (",", ":", "&" etc) into the Add-SpecialCharacters function seems to have sorted me out. I manually html'd a "&" in one of my scripts and it worked much better.

**Comment by udbellamy** [comment]
Hey, I had the same issue, I sorted that by removing commas, !, & and accentuated letters.
